From a291032fe121d46b5d3778e961e4f1d8d34cb778 Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Thu, 6 Aug 2020 15:57:49 +0530
Subject: [PATCH 25/25] Most GPIOs differ, clear "hog" misc. pinmux device

We only keep one shared native pad function and add our generic GPIOs.

Putting each single GPIO in its own group would only clutter the DT,
but a common group for all power enable pins is nice and clean.

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 arch/arm/boot/dts/lec-imx6.dtsi | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/lec-imx6.dtsi b/arch/arm/boot/dts/lec-imx6.dtsi
index fe52135a65c4..bc86f65eeaef 100644
--- a/arch/arm/boot/dts/lec-imx6.dtsi
+++ b/arch/arm/boot/dts/lec-imx6.dtsi
@@ -737,20 +737,21 @@
 	lec-imx6 {
 		pinctrl_hog: hoggrp {
 			fsl,pins = <
+				/* Native pad functions */
 				MX6QDL_PAD_GPIO_0__CCM_CLKO1    0x130b0
-				MX6QDL_PAD_NANDF_CLE__GPIO6_IO07 0x1b0b0
+				/* Pads used for GPIOs */
+				MX6QDL_PAD_GPIO_3__GPIO1_IO03   0x80000000 /* AEN# (mux: 0=SPI0, 1=I2S0) */
 				MX6QDL_PAD_ENET_TXD1__GPIO1_IO29 0x80000000 /* GPIO_INT_B */
 				MX6QDL_PAD_ENET_TXD0__GPIO1_IO30 0x80000000 /* SMARC LCD_VDD_EN */
-				MX6QDL_PAD_ENET_RXD0__GPIO1_IO27 0x80000000 /* SMARC USB0_EN_OC# (open-drain) */
+				MX6QDL_PAD_ENET_RXD0__GPIO1_IO27 0x1a8b0 /* SMARC USB0_EN_OC# (open-drain) */
 				MX6QDL_PAD_EIM_D29__GPIO3_IO29	0x1a8b0 /* SMARC USB1_EN_OC# (open-drain) */
 				MX6QDL_PAD_EIM_D30__GPIO3_IO30	0x1a8b0 /* SMARC USB2_EN_OC# (open-drain) */
 				MX6QDL_PAD_KEY_ROW2__GPIO4_IO11	0x80000000 /* SMARC SDIO_PWR_EN */
-				MX6QDL_PAD_EIM_D23__GPIO3_IO23 0x80000000
-				MX6QDL_PAD_SD3_RST__GPIO7_IO08 0x80000000
 				MX6QDL_PAD_NANDF_CS0__GPIO6_IO11 0x80000000 /* SMARC GPIO0/CAM0_PWR# */
 				MX6QDL_PAD_NANDF_CS1__GPIO6_IO14 0x80000000 /* SMARC GPIO1/CAM1_PWR# */
 				MX6QDL_PAD_NANDF_CS2__GPIO6_IO15 0x80000000 /* SMARC GPIO2/CAM0_RST# */
 				MX6QDL_PAD_NANDF_CS3__GPIO6_IO16 0x80000000 /* SMARC GPIO3/CAM1_RST# */
+				MX6QDL_PAD_NANDF_CLE__GPIO6_IO07 0x80000000 /* SMARC GPIO7/PCAM_FLD */
 			>;
 		};
 
-- 
2.17.1

